### PR TITLE
Added python setup.py test functionality

### LINKED ISSUE #2765: Add "python setup.py test" option
To codify the new convention and py.test as the test runner.

### PR DESCRIPTION
**Pull request by SaganBolliger**
Fixes #2765

Follows recommendations from https://pytest.org/latest/goodpractises.html#integrating-with-setuptools-python-setup-py-test

@scottpurdy @oxtopus Please review.

### CODE REVIEW

**Comment by SaganBolliger** [comment]
@oxtopus I've updated this PR to reflect the changes suggested in your comment on numenta/nupic.core#703. Please review.

**Comment by SaganBolliger** [comment]
@scottpurdy I've modified this PR too to reflect the naming convention you suggested in numenta/nupic.core#703.

I've also changed the test command to only run unit tests. I just wanted to double check that this is the intended behavior?

**Comment by SaganBolliger** [comment]
@scottpurdy Please review.

**Comment by scottpurdy** [comment]
I'd like to defer to @oxtopus but perhaps we should merge for now and he can do a post-merge review when he is back? :+1: if you want to go that route.

**Comment by SaganBolliger** [comment]
I'm fine with whatever you think is the best route.

File: setup.py [hunk 1]
```diff
@@ -86,6 +115,7 @@ def findRequirements():
       "nupic.swarming.jsonschema": ["*.json"],
       "nupic.datafiles": ["*.csv", "*.txt"],
     },
+    cmdclass = {'test': TestCommand},
```
Comment (scottpurdy) [inline setup.py]: double quotes

**Comment by SaganBolliger** [comment]
@scottpurdy Quotes fixed.

**Comment by scottpurdy** [comment]
:+1:

**Pull request merged**